scripts/skills/effects/rf_soul_link_effect.nut

```diff
@@ -0,0 +1,122 @@
+this.rf_soul_link_effect <- ::inherit("scripts/skills/skill", {
+	m = {
+		TransferedPart = 0.4,	// This percentage of incoming damage is prevented
+		TransferedMult = 1.0,	// This part of the prevented damage is inflicted to the nearby linked target
+
+		TransferInfo = null
+	},
+	function create()
+	{
+		this.m.ID = "effects.rf_soul_link";
+		this.m.Name = "Soul Link";
+		this.m.Description = "This character can link their soul with adjacent lesser allies. A portion of all incoming damage to Health is transfered to a random adjacent ally. Doesn't work with allies that also have Soul Link.";
+		this.m.KilledString = "Died from Soul Link";
+		this.m.Icon = "ui/perks/rf_wear_them_down.png";
+		this.m.IconMini = "rf_worn_down_effect_mini";
+		this.m.SoundOnUse = [
+			"sounds/enemies/dlc2/hexe_hex_damage_01.wav",
+			"sounds/enemies/dlc2/hexe_hex_damage_02.wav",
+			"sounds/enemies/dlc2/hexe_hex_damage_03.wav",
+			"sounds/enemies/dlc2/hexe_hex_damage_04.wav"
+		];
+		this.m.Overlay = "rf_worn_down_effect";
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.IsRemovedAfterBattle = true;
+	}
+
+	function getName()
+	{
+		if (this.hasLink()) return (this.m.Name + " (Active)");
+		return this.m.Name;
+	}
+
+	function getTooltip()
+	{
+		local tooltip = this.skill.getTooltip();
+
+		tooltip.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/health.png",
+			text = ::MSU.Text.colorGreen((this.m.TransferedPart * 100) + "%") + " of Hitpoint damage taken is transferred to a random adjacent ally that does not have Soul Link"
+		});
+
+		return tooltip;
+	}
+
+	function onDamageReceived( _attacker, _damageHitpoints, _damageArmor )
+	{
+		if (_damageHitpoints == 0) return;
+		if (!this.hasLink()) return;
+
+		local transferedDamageMult = (1.0 / (1.0 - this.m.TransferedPart)) - 1.0;
+		this.m.TransferInfo = {
+			Attacker = _attacker.weakref(),
```

Comment:
Use `Attacker = ::MSU.asWeakTableRef(_attacker)` instead of doing a manual `.weakref()`